code/go/0chain.net/smartcontract/dbs/event/provider.go

```diff
@@ -35,3 +54,69 @@ func (p *Provider) BeforeCreate(tx *gorm.DB) (err error) {
 	}
 	return
 }
+
+// ProviderHealthCheck Update given provider (providerType, providerId) with the given HealthCheckData and calculate aggregates e.g. downtime
+func (edb *EventDb) ProviderHealthCheck(providerType spenum.Provider, providerId string, data interface{}) (err error) {
+	providerModel, err	  := edb.getProviderModel(providerType)
+	if err != nil {
+		return fmt.Errorf("provider_health_check: %v", err.Error()) 
+	}
+	providerInstance, err := edb.getProviderInstance(providerModel, providerId)
+	if err != nil {
+		return fmt.Errorf("provider_health_check: %v", err.Error())
+	}
+
+	healthCheckData, ok := data.(*dbs.DbHealthCheck)
+	if !ok  {
+		return fmt.Errorf("provider_health_check: invalid data")
+	}
+	prevHealthCheck := providerInstance.LastHealthCheck
+	curHealthCheck 	:= healthCheckData.LastHealthCheck
+	diff 			:= curHealthCheck - prevHealthCheck
+	updates 		:= &Provider{}
```

Comment:
If we want to be able to merge events, this function may not work as it can only handle one provider instance each time. We need to calculate the `downtime` in smartcontract when emit the event, and let the event processor only do the update work.